### ISSUE TITLE
--Progress order of transfers not maintained

### ISSUE THREAD

**Issue by SL-Gundam** [rclone/rclone#4599]
#### What is the problem you are having with rclone?

Since 1.53.0 the order of the transfers shown for --progress is not maintained and changes almost every refresh even though its still the same 4 transfers

#### What is your rclone version (output from `rclone version`)

rclone v1.53.1
- os/arch: windows/amd64
- go version: go1.15

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Windows 10 x64 2004

####  Which cloud storage system are you using? (eg Google Drive)

Google Drive

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

C:\rclone\rclone.exe sync --bwlimit 3M --fast-list --track-renames --track-renames-strategy modtime --check-first --delete-after --delete-excluded --exclude-from "C:/rclone/Configs/Exclusions.txt" --progress --log-file "C:\Users\name\Desktop\MirrorD Log\Rclone Mirror_D.log" --log-level INFO local:D:/ cryptGDrive:D

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

n/a

**Comment by ncw** [comment]
What is supposed to happen is that the transfer started first is at the top of the list.

This was changed for 1.53

Can you paste an example of them getting out of order? Remove --progress and replace with --stats 5s and paste some stats blocks - thanks

**Comment by SL-Gundam** [comment]
As requested

```
2020/09/16 00:59:41 INFO  : Starting bandwidth limiter at 3MBytes/s
2020/09/16 00:59:42 INFO  : Encrypted drive 'cryptGDrive:D': Running all checks before starting transfers
2020/09/16 01:02:24 INFO  : Encrypted drive 'cryptGDrive:D': Making map for --track-renames
2020/09/16 01:02:24 INFO  : Encrypted drive 'cryptGDrive:D': Finished making map for --track-renames
2020/09/16 01:02:24 INFO  : Encrypted drive 'cryptGDrive:D': Checks finished, now starting transfers
2020/09/16 01:02:27 INFO  : 
Transferred:   	   10.784M / 22.854 GBytes, 0%, 3.282 MBytes/s, ETA 1h58m47s
Checks:            185453 / 185453, 100%
Transferred:            0 / 12, 0%
Elapsed time:      2m45.8s
Transferring:
 *                       Reencoding/GNABGNAB.mp4:  0% /5.429G, 864.207k/s, 1h49m44s
 *                       Reencoding/USBAUSBA.mp4:  0% /4.573G, 1.250M/s, 1h2m23s
 * Tempi/Games/Playing/So…ing_Unlimited_2.2.9.7z:  0% /1.210G, 464.002k/s, 45m32s
 * Tempi/Games/New/Hail D…or_0.8.0_Windows64.zip:  0% /1.799G, 1.453M/s, 21m4s

2020/09/16 01:02:32 INFO  : 
Transferred:   	   25.788M / 22.854 GBytes, 0%, 3.113 MBytes/s, ETA 2h5m9s
Checks:            185453 / 185453, 100%
Transferred:            0 / 12, 0%
Elapsed time:      2m50.8s
Transferring:
 * Tempi/Games/New/Hail D…or_0.8.0_Windows64.zip:  0% /1.799G, 1.040M/s, 29m23s
 *                       Reencoding/GNABGNAB.mp4:  0% /5.429G, 804.707k/s, 1h57m46s
 *                       Reencoding/USBAUSBA.mp4:  0% /4.573G, 928.112k/s, 1h25m58s
 * Tempi/Games/Playing/So…ing_Unlimited_2.2.9.7z:  0% /1.210G, 676.642k/s, 31m7s

2020/09/16 01:02:37 INFO  : 
Transferred:   	   40.729M / 22.854 GBytes, 0%, 3.066 MBytes/s, ETA 2h7m
Checks:            185453 / 185453, 100%
Transferred:            0 / 12, 0%
Elapsed time:      2m55.8s
Transferring:
 * Tempi/Games/New/Hail D…or_0.8.0_Windows64.zip:  0% /1.799G, 938.865k/s, 33m16s
 *                       Reencoding/GNABGNAB.mp4:  0% /5.429G, 789.528k/s, 1h59m57s
 *                       Reencoding/USBAUSBA.mp4:  0% /4.573G, 861.521k/s, 1h32m32s
 * Tempi/Games/Playing/So…ing_Unlimited_2.2.9.7z:  0% /1.210G, 714.820k/s, 29m22s

2020/09/16 01:02:42 INFO  : 
Transferred:   	   55.732M / 22.854 GBytes, 0%, 3.048 MBytes/s, ETA 2h7m39s
Checks:            185453 / 185453, 100%
Transferred:            0 / 12, 0%
Elapsed time:       3m0.8s
Transferring:
 *                       Reencoding/GNABGNAB.mp4:  0% /5.429G, 781.298k/s, 2h1m8s
 *                       Reencoding/USBAUSBA.mp4:  0% /4.573G, 835.669k/s, 1h35m20s
 * Tempi/Games/Playing/So…ing_Unlimited_2.2.9.7z:  1% /1.210G, 728.770k/s, 28m43s
 * Tempi/Games/New/Hail D…or_0.8.0_Windows64.zip:  0% /1.799G, 888.177k/s, 35m5s

2020/09/16 01:02:47 INFO  : 
Transferred:   	   70.705M / 22.854 GBytes, 0%, 3.037 MBytes/s, ETA 2h8m3s
Checks:            185453 / 185453, 100%
Transferred:            0 / 12, 0%
Elapsed time:       3m5.8s
Transferring:
 *                       Reencoding/USBAUSBA.mp4:  0% /4.573G, 815.570k/s, 1h37m36s
 * Tempi/Games/Playing/So…ing_Unlimited_2.2.9.7z:  1% /1.210G, 743.192k/s, 28m5s
 * Tempi/Games/New/Hail D…or_0.8.0_Windows64.zip:  1% /1.799G, 855.465k/s, 36m22s
 *                       Reencoding/GNABGNAB.mp4:  0% /5.429G, 775.370k/s, 2h1m58s

2020/09/16 01:02:52 INFO  : 
Transferred:   	   85.709M / 22.854 GBytes, 0%, 3.030 MBytes/s, ETA 2h8m14s
Checks:            185453 / 185453, 100%
Transferred:            0 / 12, 0%
Elapsed time:      3m10.8s
Transferring:
 *                       Reencoding/GNABGNAB.mp4:  0% /5.429G, 773.375k/s, 2h2m12s
 *                       Reencoding/USBAUSBA.mp4:  0% /4.573G, 804.032k/s, 1h38m55s
 * Tempi/Games/Playing/So…ing_Unlimited_2.2.9.7z:  1% /1.210G, 748.526k/s, 27m47s
 * Tempi/Games/New/Hail D…or_0.8.0_Windows64.zip:  1% /1.799G, 832.925k/s, 37m16s

2020/09/16 01:02:57 INFO  : 
Transferred:   	  100.743M / 22.854 GBytes, 0%, 3.027 MBytes/s, ETA 2h8m18s
Checks:            185453 / 185453, 100%
Transferred:            0 / 12, 0%
Elapsed time:      3m15.8s
Transferring:
 *                       Reencoding/GNABGNAB.mp4:  0% /5.429G, 771.563k/s, 2h2m25s
 *                       Reencoding/USBAUSBA.mp4:  0% /4.573G, 794.251k/s, 1h40m3s
 * Tempi/Games/Playing/So…ing_Unlimited_2.2.9.7z:  1% /1.210G, 755.929k/s, 27m26s
 * Tempi/Games/New/Hail D…or_0.8.0_Windows64.zip:  1% /1.799G, 813.538k/s, 38m5s

2020/09/16 01:03:02 INFO  : 
Transferred:   	  115.778M / 22.854 GBytes, 0%, 3.024 MBytes/s, ETA 2h8m20s
Checks:            185453 / 185453, 100%
Transferred:            0 / 12, 0%
Elapsed time:      3m20.8s
Transferring:
 *                       Reencoding/GNABGNAB.mp4:  0% /5.429G, 770.659k/s, 2h2m28s
 *                       Reencoding/USBAUSBA.mp4:  0% /4.573G, 787.090k/s, 1h40m53s
 * Tempi/Games/Playing/So…ing_Unlimited_2.2.9.7z:  2% /1.210G, 757.790k/s, 27m17s
 * Tempi/Games/New/Hail D…or_0.8.0_Windows64.zip:  1% /1.799G, 801.056k/s, 38m35s

2020/09/16 01:03:07 INFO  : 
Transferred:   	  130.720M / 22.854 GBytes, 1%, 3.020 MBytes/s, ETA 2h8m25s
Checks:            185453 / 185453, 100%
Transferred:            0 / 12, 0%
Elapsed time:      3m25.8s
Transferring:
 *                       Reencoding/GNABGNAB.mp4:  0% /5.429G, 771.965k/s, 2h2m11s
 *                       Reencoding/USBAUSBA.mp4:  0% /4.573G, 779.734k/s, 1h41m45s
 * Tempi/Games/Playing/So…ing_Unlimited_2.2.9.7z:  2% /1.210G, 762.647k/s, 27m1s
 * Tempi/Games/New/Hail D…or_0.8.0_Windows64.zip:  1% /1.799G, 790.259k/s, 39m2s

2020/09/16 01:03:12 INFO  : 
Transferred:   	  145.786M / 22.854 GBytes, 1%, 3.019 MBytes/s, ETA 2h8m22s
Checks:            185453 / 185453, 100%
Transferred:            0 / 12, 0%
Elapsed time:      3m30.8s
Transferring:
 *                       Reencoding/GNABGNAB.mp4:  0% /5.429G, 769.507k/s, 2h2m29s
 *                       Reencoding/USBAUSBA.mp4:  0% /4.573G, 778.121k/s, 1h41m53s
 * Tempi/Games/Playing/So…ing_Unlimited_2.2.9.7z:  2% /1.210G, 762.450k/s, 26m57s
 * Tempi/Games/New/Hail D…or_0.8.0_Windows64.zip:  2% /1.799G, 785.750k/s, 39m11s

2020/09/16 01:03:17 INFO  : 
Transferred:   	  160.789M / 22.854 GBytes, 1%, 3.018 MBytes/s, ETA 2h8m22s
Checks:            185453 / 185453, 100%
Transferred:            0 / 12, 0%
Elapsed time:      3m35.8s
Transferring:
 *                       Reencoding/USBAUSBA.mp4:  0% /4.573G, 773.320k/s, 1h42m26s
 * Tempi/Games/Playing/So…ing_Unlimited_2.2.9.7z:  3% /1.210G, 765.899k/s, 26m45s
 * Tempi/Games/New/Hail D…or_0.8.0_Windows64.zip:  2% /1.799G, 779.145k/s, 39m26s
 *                       Reencoding/GNABGNAB.mp4:  0% /5.429G, 771.079k/s, 2h2m9s

2020/09/16 01:03:22 INFO  : 
Transferred:   	  175.793M / 22.854 GBytes, 1%, 3.016 MBytes/s, ETA 2h8m20s
Checks:            185453 / 185453, 100%
Transferred:            0 / 12, 0%
Elapsed time:      3m40.8s
Transferring:
 *                       Reencoding/GNABGNAB.mp4:  0% /5.429G, 768.656k/s, 2h2m27s
 *                       Reencoding/USBAUSBA.mp4:  0% /4.573G, 773.685k/s, 1h42m18s
 * Tempi/Games/Playing/So…ing_Unlimited_2.2.9.7z:  3% /1.210G, 764.642k/s, 26m42s
 * Tempi/Games/New/Hail D…or_0.8.0_Windows64.zip:  2% /1.799G, 777.862k/s, 39m25s

```

**Comment by SL-Gundam** [comment]
I think the issue is that i use --check-first. Which causes 4 transfers to start simultaneously. Since they all have the same start time the order is messed up until three of the four are done and there are actual start time differences for all transfers

**Comment by ncw** [comment]
Thanks for the log demonstrating the problem - it is quite clear there.

> I think the issue is that i use --check-first. Which causes 4 transfers to start simultaneously. Since they all have the same start time the order is messed up until three of the four are done and there are actual start time differences for all transfers

The times used for sorting have nano second resolution... However I think Windows time sources don't have that much resolution, I think only 1mS maybe. The go runtime could easily queue 4 transfers in 1mS...

I guess you don't much care about which order they come in, as long as they are consistent.

Try this which sorts the transfers by name if the times are equal. This should stop them jumping around hopefully!

[v1.54.0-beta.4775.94e48bca0.fix-4599-stats-order](https://beta.rclone.org/branch/fix-4599-stats-order/v1.54.0-beta.4775.94e48bca0.fix-4599-stats-order/) on branch [fix-4599-stats-order](https://github.com/rclone/rclone/tree/fix-4599-stats-order) (uploaded in 15-30 mins)

**Comment by SL-Gundam** [comment]
Seems to working just fine with this update
thanks

**Comment by ncw** [comment]
Thanks for testing.

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.53.2
